book/src/03_5_http_client.md

```diff
@@ -0,0 +1,159 @@
+# HTTP Client
+Next, we'll write a small client that retrieves data over an HTTP connection to the internet.
+
+Before jumping to the exercise, let's explore how Wi-Fi works in `no_std` Rust for Espressif devices.
+
+## Wi-Fi Ecosystem
+
+Wi-Fi support comes in the [`esp-wifi` crate][esp-wifi]. The `esp-wifi` is home to the Wi-Fi, Bluetooth and ESP-NOW driver implementations for `no_std` Rust.
+Check the repository README for current support, limitations and usage details.
+
+There are some other relevant crates, on which `esp-wifi` depends on:
+- [`embedded-svc`][embedded-svc]: Contains traits for features such as wifi, networking, HTTPD, and logging.
```

Comment:
While this is generally true we currently only have a bare-metal implementation for the `wifi` module (living in esp-wifi) - so maybe just mention that?

off-topic: I thought about implementing some other traits (in other crates) for bare-metal. I don't agree with all of them in general and the for some parts with a few details t.b.h. - but that discussion doesn't belong here - will raise that question in our next community meeting (if I don't forget about it)

Reply:
Just updated this information in https://github.com/esp-rs/no_std-training/pull/7/commits/88bbba996de35689d2dfbf3c3036f971d4d2be45